server/service/osquery.go

```diff
@@ -1115,6 +1115,24 @@ func (svc *Service) SubmitDistributedQueryResults(
 	}
 
 	if len(labelResults) > 0 {
+		// Force clear results for labels that do not apply to the host anymore.
+		//
+		// There could be a timing bug where:
+		// 1. Host receives a "team label" query to run (distributed/read).
+		// 2. Host is transferred to another team (all its label/policy membership are cleared).
+		// 3. Fleet receives distributed/write corresponding to (1) which includes the result for
+		//    the label of the old team.
+		hostLabelQueries, err := svc.ds.LabelQueriesForHost(ctx, host)
```

Comment:
Was trying to think through how to make this faster given that it's a hot path, but given that flipping the query to "give me labels back that I supplied but shouldn't be on the host" allows the host to potentially DoS the server with e.g. 65k labels, maybe this is the best we can do?